var/spack/repos/builtin/packages/py-jupyter-server-proxy/package.py

```diff
@@ -0,0 +1,28 @@
+# Copyright 2013-2023 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack.package import *
+
+
+class PyJupyterServerProxy(PythonPackage):
+    """
+    Jupyter Server Proxy lets you run arbitrary external processes
+    (such as RStudio, Shiny Server, Syncthing, PostgreSQL, Code Server, etc)
+    alongside your notebook server and provide authenticated web access to them
+    using a path like /rstudio next to others like /lab.
+    """
+
+    homepage = "https://github.com/jupyterhub/jupyter-server-proxy"
+    pypi = "jupyter-server-proxy/jupyter-server-proxy-3.2.2.tar.gz"
+
+    version("3.2.2", sha256="54690ea9467035d187c930c599e76065017baf16e118e6eebae0d3a008c4d946")
+
+    depends_on("py-jupyter-packaging@0.7.9:0.7", type="build")
```

Comment:
```suggestion
    depends_on("py-jupyter-packaging7@0.7.9:0.7", type="build")
```
We had a lot of concretization issues with jupyter-packaging and ended up splitting it by version.